edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/IKubernetesDeploymentMapper.cs

```diff
@@ -7,7 +7,7 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment
 
     public interface IKubernetesDeploymentMapper
     {
-        V1Deployment CreateDeployment(IModuleIdentity identity, KubernetesModule module, IDictionary<string, string> labels);
+        V1Deployment CreateDeployment(IModuleIdentity identity, KubernetesModule module, IDictionary<string, string> labels, EdgeDeploymentDefinition edgeDeploymentDefinition);
```

Comment:
I don't think it is a good idea to pass the whole EdgeDeploymentDefinition in each method for mappers. 
I think maybe we can create a context for each method `Create{Resource}()` and this context will contain data to make a conversion to `{Resource}`. It will help us:
1. pass additional data into methods in the future.
2. we can even make a class hierarchy for these contexts

Reply:
Or we can make this owner information to be a part of KubernetesModule. And in the long run this KubernetesModule can become a CRD object to operate on.
```csharp
    public class KubernetesModule
    {
        ...
        // TODO initialize via constructor
        public KubernetesModuleOwner Owner { get; }
        ...
    }

    public class KubernetesModuleOwner
    {
        // TODO initialize via constructor
        public string ApiVersion { get; }
        public string Kind { get; }
        public string Name { get; }
        public string Namespace { get; }
        public string Uid { get; }
    }
```

Reply:
The KubernetesModule and KubernetesConfig contains fields that are specified by IoTHub. They are pretty much clones of DockerModule and DockerConfig. However, CRD objects (aka edgeDeploymentDefinition) represent Kubernetes objects created at deployment time only. I am reluctant to mix the two at the moment.

Reply:
Good suggestion. Please review latest changes!